### DISCUSSION TITLE
extracting paragraphs from pdf

### DISCUSSION THREAD

**Discussion by vasnt** [BurntSushi/ripgrep#2398] [Q&A] [answered]
by multiple keywords (from a given list) search , Is it possible to extract (copy) matching paragraphs & lines from pdf/s  & store them in any text/markdown file ? How? along with page/file name reference.

**Comment by BurntSushi** [comment, marked as answer]
ripgrep is a tool for searching plain text. It doesn't know how to read PDFs because PDFs are not plain text. Your question sounds like one for someone experienced in PDF text extraction. I am not that person.
